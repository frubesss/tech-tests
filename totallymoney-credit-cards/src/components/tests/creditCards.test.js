import React from 'react';
import renderer from 'react-test-renderer'
import {mount} from 'enzyme';
import CreditCards from "../creditCards";

describe('Credit Cards Component', () => {

    let wrapper;

    const mockedCustomer = {
            employment: 'fullTimeEmployment',
            annualIncome: 1
        };

    const mockedRickStudentCustomer = {
            employment: 'student',
            annualIncome: 16001
        };

    const mockedFullTimeCustomer = {
        employment: 'fullTimeEmployment',
        annualIncome: 16001
    };

    beforeEach(() => {
        wrapper = mount(<CreditCards customer={mockedCustomer}/>);
    });

    it('Snapshot Test', () => {
        const tree = renderer.create(wrapper).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should always offer anywhere card to the customer', () => {
        expect(wrapper.find('.title').text()).toBe('Anywhere Card');
    });

    it('should offer student card to a student customer', () => {
        wrapper = mount(<CreditCards customer={mockedRickStudentCustomer}/>);
        expect(wrapper.find('.title').first().text()).toBe('Student Life Card');
    });

    it('should offer liquid credit card to customer with income over 16000', () => {
        wrapper = mount(<CreditCards customer={mockedFullTimeCustomer}/>);
        expect(wrapper.find('.title').last().text()).toBe('Liquid Card');
    });

    it('should offer all cards to student customer with income over 16000', () => {
        wrapper = mount(<CreditCards customer={mockedRickStudentCustomer}/>);
        expect(wrapper.find('.card')).toHaveLength(3);
    });


});

