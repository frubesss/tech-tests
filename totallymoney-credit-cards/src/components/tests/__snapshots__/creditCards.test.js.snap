// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Credit Cards Component Snapshot Test 1`] = `
<div>
  <div>
    <div>
      <div
        className="card-wrapper"
      >
        <div
          className="card"
        >
          <p
            className="title"
          >
            Anywhere Card
          </p>
          <p
            className="apr"
          >
            Apr: 
            <strong>
              33.9
              %
            </strong>
          </p>
          <p
            className="b-t"
          >
            Balance Transfer Offer Duration: 
            <strong>
              0
               Months
            </strong>
          </p>
          <p
            className="purchase"
          >
            Purchase Offer Duration: 
            <strong>
              0
               Months
            </strong>
          </p>
          <p
            className="credit"
          >
            Credit Available: 
            <strong>
              Â£
              300
            </strong>
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
